src/renderer/uia/UiaRenderer.cpp

```diff
@@ -0,0 +1,414 @@
+[[nodiscard]] HRESULT UiaEngine::PaintCursor(const IRenderEngine::CursorOptions& /*options*/) noexcept
+{
+    return S_FALSE;
+}
+
+// Routine Description:
+// - Updates the default brush colors used for drawing
+// - Not currently used by UiaEngine.
+// Arguments:
+// - colorForeground - <unused>
+// - colorBackground - <unused>
+// - legacyColorAttribute - <unused>
+// - isBold - <unused>
+// - isSettingDefaultBrushes - <unused>
+// Return Value:
+// - S_FALSE since we do nothing
+[[nodiscard]] HRESULT UiaEngine::UpdateDrawingBrushes(COLORREF const /*colorForeground*/,
+                                                      COLORREF const /*colorBackground*/,
+                                                      const WORD /*legacyColorAttribute*/,
+                                                      const bool /*isBold*/,
+                                                      bool const /*isSettingDefaultBrushes*/) noexcept
+{
+    return S_FALSE;
+}
+
+// Routine Description:
+// - Updates the font used for drawing
+// Arguments:
+// - pfiFontInfoDesired - <unused>
+// - fiFontInfo - <unused>
+// Return Value:
+// - S_FALSE since we do nothing
+[[nodiscard]] HRESULT UiaEngine::UpdateFont(const FontInfoDesired& /*pfiFontInfoDesired*/, FontInfo& /*fiFontInfo*/) noexcept
+{
+    // TODO CARLOS: changing the size may be useful
+    return E_NOTIMPL;
+}
+
+// Routine Description:
+// - Sets the DPI in this renderer
+// - Not currently used by UiaEngine.
+// Arguments:
+// - iDpi - DPI
+// Return Value:
+// - S_OK
+[[nodiscard]] HRESULT UiaEngine::UpdateDpi(int const /*iDpi*/) noexcept
+{
+    // TODO CARLOS: changing the size may be useful
+    return E_NOTIMPL;
+}
+
+// Method Description:
+// - This method will update our internal reference for how big the viewport is.
+//      Does nothing for DX.
+// Arguments:
+// - srNewViewport - The bounds of the new viewport.
+// Return Value:
+// - HRESULT S_OK
+[[nodiscard]] HRESULT UiaEngine::UpdateViewport(const SMALL_RECT /*srNewViewport*/) noexcept
+{
+    // TODO CARLOS: not sure how to handle resizing just yet
+    return E_NOTIMPL;
+}
+
+// Routine Description:
+// - Currently unused by this renderer
+// Arguments:
+// - pfiFontInfoDesired - <unused>
+// - pfiFontInfo - <unused>
+// - iDpi - <unused>
+// Return Value:
+// - S_FALSE
+[[nodiscard]] HRESULT UiaEngine::GetProposedFont(const FontInfoDesired& /*pfiFontInfoDesired*/,
+                                                 FontInfo& /*pfiFontInfo*/,
+                                                 int const /*iDpi*/) noexcept
+{
+    // TODO CARLOS: changing the size may be useful
+    return S_FALSE;
+}
+
+// Routine Description:
+// - Gets the area that we currently believe is dirty within the character cell grid
+// - Not currently used by UiaEngine.
+// Arguments:
+// - <none>
+// Return Value:
+// - Rectangle describing dirty area in characters.
+[[nodiscard]] SMALL_RECT UiaEngine::GetDirtyRectInChars() noexcept
+{
+    return {};

```

Comment:
Uh. Probably return Viewport::Empty().ToInclusive(), not just default construct.